### PR TITLE
TMA-705 deprecated delete_projects and delete_extra param and added …

### PR DESCRIPTION
**Pull request by Hahihula**
…new param delete_mode

### CODE REVIEW

File: lib/gooddata/lcm/actions/associate_clients.rb [hunk 1]
```diff
@@ -46,7 +49,21 @@ class AssociateClients < BaseAction
       ]
 
       class << self
+        DELETE_MODES = %w(
+          none
+          delete_projects
+          delete_extra
+        )
         def call(params)
+          if params.delete_projects && !params.delete_extra
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Calls 'params.delete_extra' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/lcm/actions/associate_clients.rb [hunk 2]
```diff
@@ -46,7 +49,21 @@ class AssociateClients < BaseAction
...
+            GoodData.logger.warn("Parameter `delete_projects` is set to true and parameter `delete_extra` is set to false (which is default) this action will do nothing!")
+            return
+          end
+
+          unless DELETE_MODES.include?(params.delete_mode)
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Calls 'params.delete_mode' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/100/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/918/ : SUCCESS in 39s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1187/ : FAILURE in 53s

**Comment by Hahihula** [comment]
extended test

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/108/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/925/ : SUCCESS in 50s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1194/ : FAILURE in 59s

**Comment by Hahihula** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/115/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1377/ : SUCCESS in 57m 27s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1143/ : UNSTABLE in 25m 42s

**Pull request closed without merging**